tests/robot/suites/traffic/tap_int_traffic/tap_int_traffic.robot

```diff
@@ -0,0 +1,125 @@
+    ${actual_state}=    vpp_term: Check TAP interface State    agent_vpp_2    ${NAME_VPP2_TAP1}    mac=${MAC_VPP2_TAP1}    ipv4=${IP_VPP2_TAP1}/${PREFIX}    state=${UP_STATE}
+
+Check Ping Between VPP2 And linux_VPP2_TAP1 Interface
+    linux: Check Ping    node=agent_vpp_2    ip=${IP_VPP2_TAP1}
+    vpp_term: Check Ping    node=agent_vpp_2    ip=${IP_LINUX_VPP2_TAP1}
+
+Add VPP2_memif1 Interface
+    vpp_term: Interface Not Exists    node=agent_vpp_2    mac=${MAC_VPP2_MEMIF1}
+    vpp_ctl: Put Memif Interface With IP    node=agent_vpp_2    name=${NAME_VPP2_MEMIF1}    mac=${MAC_VPP2_MEMIF1}    master=false    id=1    ip=${IP_VPP2_MEMIF1}    prefix=24    socket=memif.sock
+    vpp_term: Interface Is Created    node=agent_vpp_1    mac=${MAC_VPP1_MEMIF1}
+
+Check Ping From VPP1 To VPP2_memif1
+    vpp_term: Check Ping    node=agent_vpp_1    ip=${IP_VPP2_MEMIF1}
+
+Check Ping From VPP2 To VPP1_memif1
+    vpp_term: Check Ping    node=agent_vpp_2    ip=${IP_VPP1_MEMIF1}
+
+Ping From VPP1 Linux To VPP2_TAP1 And LINUX_VPP2_TAP1 Should Not Pass
+    ${status1}=    Run Keyword And Return Status    linux: Check Ping    node=agent_vpp_1    ip=${IP_VPP2_TAP1}
+    ${status2}=    Run Keyword And Return Status    linux: Check Ping    node=agent_vpp_1    ip=${IP_LINUX_VPP2_TAP1}
+    Should Be Equal As Strings    ${status1}    False
+    Should Be Equal As Strings    ${status2}    False
+
+Ping From VPP2 Linux To VPP1_TAP1 And LINUX_VPP1_TAP1 Should Not Pass
+    ${status1}=    Run Keyword And Return Status    linux: Check Ping    node=agent_vpp_2    ip=${IP_VPP1_TAP1}
+    ${status2}=    Run Keyword And Return Status    linux: Check Ping    node=agent_vpp_2    ip=${IP_LINUX_VPP1_TAP1}
+    Should Be Equal As Strings    ${status1}    False
+    Should Be Equal As Strings    ${status2}    False
+
+Add Static Route From VPP1 Linux To VPP1
+    linux: Add Route    node=agent_vpp_1    destination_ip=${IP_VPP2_TAP1_NETWORK}    prefix=${PREFIX}    next_hop_ip=${IP_VPP1_TAP1}
+
+Add Static Route From VPP1 To VPP2
+    vpp_term: Add Route    node=agent_vpp_1    destination_ip=${IP_VPP2_TAP1_NETWORK}    prefix=${PREFIX}    next_hop_ip=${IP_VPP2_MEMIF1}
+
+Add Static Route From VPP2 Linux To VPP2
+    linux: Add Route    node=agent_vpp_2    destination_ip=${IP_VPP1_TAP1_NETWORK}    prefix=${PREFIX}    next_hop_ip=${IP_VPP2_TAP1}
+
+Add Static Route From VPP2 To VPP1
+    vpp_term: Add Route    node=agent_vpp_2    destination_ip=${IP_VPP1_TAP1_NETWORK}    prefix=${PREFIX}    next_hop_ip=${IP_VPP1_MEMIF1}
+
+Check Ping From VPP1 Linux To VPP2_TAP1 And LINUX_VPP2_TAP1
+    linux: Check Ping    node=agent_vpp_1    ip=${IP_VPP2_TAP1}
+    linux: Check Ping    node=agent_vpp_1    ip=${IP_LINUX_VPP2_TAP1}
+
+Check Ping From VPP2 Linux To VPP1_TAP1 And LINUX_VPP1_TAP1
+    linux: Check Ping    node=agent_vpp_2    ip=${IP_VPP1_TAP1}
+    linux: Check Ping    node=agent_vpp_2    ip=${IP_LINUX_VPP1_TAP1}
+
+#*** Keywords ***
```

Comment:
add resync test, ( remove all nodes, and start them again and test all pings work )